internal/resource/url.go

```diff
@@ -422,14 +423,28 @@ func (f *Fetcher) fetchFromS3(u url.URL, dest s3target, opts FetchOptions) error
 
 	sess.Config.Region = aws.String(region)
 
+	// Determine if the S3 URL is an access point ARN
+	var bucket, key string
+	if arn.IsARN(u.Opaque) {
+		urlSplit := strings.Split(u.Opaque, "/")
+		if len(urlSplit) < 2 {
+			return fmt.Errorf("failed to parse S3 access point from ARN %s, expected valid access point format", u.Opaque)
+		}
+		bucket = strings.Join(urlSplit[:2], "/")
```

Comment:
Since I modified this to support a new `arn` URL scheme, this should now allow ARNs for an S3 object directly. I also added comments about the expected format of the ARN resource and official AWS documentation that supports those expectations.